src/Bicep.LangServer/Handlers/BicepTextDocumentSyncHandler.cs

```diff
@@ -20,26 +21,38 @@ public class BicepTextDocumentSyncHandler : TextDocumentSyncHandlerBase
     {
         private readonly ICompilationManager compilationManager;
         private readonly IBicepConfigChangeHandler bicepConfigChangeHandler;
+        private readonly IParamsCompilationManager paramsCompilationManager;
 
-        public BicepTextDocumentSyncHandler(ICompilationManager compilationManager, IBicepConfigChangeHandler bicepConfigChangeHandler)
+        public BicepTextDocumentSyncHandler(ICompilationManager compilationManager, IParamsCompilationManager paramsCompilationManager, IBicepConfigChangeHandler bicepConfigChangeHandler) //add param manager 
         {
             this.bicepConfigChangeHandler = bicepConfigChangeHandler;
             this.compilationManager = compilationManager;
+            this.paramsCompilationManager = paramsCompilationManager;
         }
 
         public override TextDocumentAttributes GetTextDocumentAttributes(DocumentUri uri)
         {
             return new TextDocumentAttributes(uri, LanguageConstants.LanguageId);
         }
 
-        public override Task<Unit> Handle(DidChangeTextDocumentParams request, CancellationToken token)
+        public override Task<Unit> Handle(DidChangeTextDocumentParams request, CancellationToken token) //this?? if this config where the change at?
         {
             // we have full sync enabled, so apparently first change is the whole document
             var contents = request.ContentChanges.First().Text;
 
             var documentUri = request.TextDocument.Uri;
 
-            this.compilationManager.UpsertCompilation(documentUri, request.TextDocument.Version, contents);
+            var fileExtension = Path.GetExtension(request.TextDocument.Uri.Path);
+
+            if (fileExtension == ".bicep")
+            {
+                this.compilationManager.UpsertCompilation(documentUri, request.TextDocument.Version, contents);
+            }
```

Comment:
Instead of checking for the `.bicep` extension, I would suggest treating everything that is *not* a bicep config file or a `.bicepparam` file as a Bicep file. Doing so would match the current behavior and keep Intellisense working in unsaved buffers.